### PR TITLE
Remove temp file after cmd execution

### LINKED ISSUE #9830: `windows/manage/run_as` leaves files on disk with every run
## Steps to reproduce

```
msf5 post(windows/manage/run_as) > set CMD ver
CMD => ver
msf5 post(windows/manage/run_as) > set CMDOUT true
CMDOUT => true
msf5 post(windows/manage/run_as) > set USER Administrator
USER => root
msf5 post(windows/manage/run_as) > set PASSWORD reallygoodpassword
PASSWORD => toor
msf5 post(windows/manage/run_as) > set SESSION 1
SESSION => 1
msf5 post(windows/manage/run_as) > set DOMAIN .
DOMAIN => .
msf5 post(windows/manage/run_as) > run
```

Note that when you run the command with CMDOUT set to True, the output is written to a randomly-generated text file in `c:\windows\temp`:

```
msf5 post(windows/manage/run_as) > run

[*] Executing CreateProcessWithLogonW...
[+] Process started successfully, PID: 900
[*] Command Run: cmd.exe /c ver > C:\Windows\Temp\AhEzUrlu.txt
[*] Command output:

Microsoft Windows [Version 10.0.16299.309]

[*] Post module execution completed
```

After completion, the file is left on disk:

```
msf5 post(windows/manage/run_as) > sessions -i 1 -C 'dir C:\\Windows\\Temp\\nzrzpzSv.txt'
[*] Running 'dir C:\\Windows\\Temp\\nzrzpzSv.txt' on meterpreter session 1 (192.168.108.217)
100666/rw-rw-rw-  46  fil  2018-04-06 13:44:08 -0500  C:\Windows\Temp\nzrzpzSv.txt
```

## Expected behavior

The command should run and return output (which it does).  But in the act of returning output, it shouldn't leave behind a text file on disk

## Current behavior

Evidence is left on disk.

**NOTE**: Counterinuitively, in the event that you turn CMDOUT off, a zero-byte file is written to disk and left behind.

### Metasploit version

```
msf5 post(windows/manage/run_as) > version
Framework: 5.0.0-dev-4dc36c0591
Console  : 5.0.0-dev-4dc36c0591
```

### I installed Metasploit with:
- [X] Source install (please specify ruby version)

```
msf5 post(windows/manage/run_as) > ruby -v
[*] exec: ruby -v

ruby 2.4.3p205 (2017-12-14 revision 61247) [x86_64-darwin17]
```

### OS

Mac OS X 10.13.3 (fully patched)

### PR DESCRIPTION
**Pull request by Auxilus**
resolves #9830

### CODE REVIEW

**Comment by asoto-r7** [comment]
Hey @Auxilus, I appreciate the quick response.  I'll run a few quick tests today and get this merged.

**Comment by Auxilus** [comment]
@asoto-r7 always up for newbie-friendlies... 😅

**Comment by wvu-r7** [comment]
https://blog.github.com/2013-05-14-closing-issues-via-pull-requests/

File: modules/post/windows/manage/run_as.rb [hunk 1]
```diff
@@ -121,5 +121,7 @@ def run
       vprint_status("Thread Id: #{pi[:thread_id]}")
       print_status("Command output:\r\n#{tmpout}") unless tmpout.nil?
     end
+    print_status("Cleaning up...")
```
Comment (wvu-r7) [inline modules/post/windows/manage/run_as.rb]: Newline before this. I also suggest telling the user what file you removed.

**Comment by wvu-r7** [comment]
> Output is not returned by default, by setting CMDOUT to **false** output will be redirected to a temp file and read back in to display.

I believe that should say **true** in the module description.

File: modules/post/windows/manage/run_as.rb [hunk 2]
```diff
@@ -121,5 +121,7 @@ def run
...
+    rm_f(outpath)
```
Comment (wvu-r7) [inline modules/post/windows/manage/run_as.rb]: Arguably, you shouldn't even be writing the file if `CMDOUT` isn't set. That should be fixed.

**Comment by asoto-r7** [comment]
Landed with adjustments:  be18930f

**Comment by allrosenthal-r7** [comment]
# Release Notes

This fix cleans up a text file left behind by the post/windows/manage/run_as module. The module no longer leaves evidence on the disk.

**Pull request merged**